implementation/src/main/java/io/smallrye/config/EnhancedConfig.java

```diff
@@ -0,0 +1,162 @@
+package io.smallrye.config;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.function.IntFunction;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.config.spi.Converter;
+
+import io.smallrye.common.annotation.Experimental;
+
+public interface EnhancedConfig extends Config, Serializable {
```

Comment:
I think maybe something like `ConfigView` is more appropriate. We don't want to present this as "an enhanced version of MP Config", but rather, "the basic SmallRye configuration view".